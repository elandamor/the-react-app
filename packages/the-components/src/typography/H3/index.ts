import {
  shouldForwardProp,
  space,
  styled,
  StyledSystemProps,
  typography,
} from '@app/theme';

/**
 * @render react
 * @name H3 component
 * @description Heading level 3.
 * @example
 *  <H3>Heading</H3>
 */

const H3 = styled('h3', { shouldForwardProp })<StyledSystemProps>`
  ${space};
  ${typography};
`;

H3.defaultProps = {
  color: 'text.default',
  fontSize: [6, 7],
};

export default H3;
