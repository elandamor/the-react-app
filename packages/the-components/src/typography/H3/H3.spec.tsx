import { render } from '@testing-library/react';
import React from 'react';
import H3 from './index';

describe('H3', () => {
  it('renders without crashing', () => {
    const { container } = render(<H3>Heading</H3>);
    expect(container.firstChild?.nodeName.toLowerCase()).toBe('h3');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correct text', () => {
    const { getByText } = render(<H3>Heading</H3>);
    expect(getByText('Heading')).toBeTruthy();
  });
});
