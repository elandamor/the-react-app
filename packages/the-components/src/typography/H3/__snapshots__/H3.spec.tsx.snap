// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H3 renders without crashing 1`] = `
.emotion-0 {
  font-size: 48px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 64px;
  }
}

<h3
  class="emotion-0"
>
  Heading
</h3>
`;
